zerver/lib/actions.py

```diff
@@ -1956,26 +1955,16 @@ def do_send_messages(
         else:
             user_list = list(user_ids)
 
-        user_data_objects: List[UserMessageNotificationsData] = []
+        online_push_user_ids = list(send_request.online_push_user_ids)
+        stream_push_user_ids = list(send_request.stream_push_user_ids)
+        stream_email_user_ids = list(send_request.stream_email_user_ids)
+        wildcard_mention_user_ids = list(send_request.wildcard_mention_user_ids)
+        muted_sender_user_ids = list(send_request.muted_sender_user_ids)
+
+        users: List[Dict[str, Union[int, List[str]]]] = []
         for user_id in user_list:
             flags = user_flags.get(user_id, [])
-            wildcard_mention_notify = (
-                user_id in send_request.wildcard_mention_user_ids and "wildcard_mentioned" in flags
-            )
-            user_data_objects.append(
-                UserMessageNotificationsData(
-                    id=user_id,
-                    flags=flags,
-                    mentioned=("mentioned" in flags),
```

Comment:
`mentioned` seems to disappear in this commit.  Is that intentional?

Reply:
Yep. Because we can calculate it from just the `flags`, we can remove it entirely from this dict. This essentially reverts 217927502035cef8d7f656a6a8df3981dffbd4b7.

Reply:
Oh! I had forgotten to pop the `mentioned` field. I did that and also added a comment there.